uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -3840,11 +3673,13 @@ private function getBalanceChangesForPeriodEntitlement($leavePeriodEntitlement)
     return $record;
   }
 
-  private function calculateAmountForDateInDays($leaveRequest, DateTime $date) {
-    return LeaveBalanceChange::calculateAmountForDate($leaveRequest, $date, $this->dateAmountDeductionService);
+  private function calculateAmountForDateInDays(LeaveRequest $leaveRequest, DateTime $date, $amount) {
+    $dayAmountDeductionService = $this->createLeaveDateDaysAmountDeductionServiceMock($amount);
+    return LeaveBalanceChange::calculateAmountForDate($leaveRequest, $date, $dayAmountDeductionService);
   }
 
-  private function calculateAmountForDateInHours($leaveRequest, DateTime $date) {
-    return LeaveBalanceChange::calculateAmountForDate($leaveRequest, $date, $this->hoursAmountDeductionService);
+  private function calculateAmountForDateInHours(LeaveRequest $leaveRequest, DateTime $date, $amount) {
+    $hoursAmountDeductionService = $this->createLeaveDateHoursAmountDeductionServiceMock($amount);
+    return LeaveBalanceChange::calculateAmountForDate($leaveRequest, $date, $hoursAmountDeductionService);
```

Comment:
You don't need two separate mocks. You only need one that mocks the interface instead of the concrete classes. In these tests, we don't really care about hours or days. All we care is that the `calculateAmountForDate()` method will call the deductionService once